templates/shared/forms/interactive/openstack-cost-calculator.html

```diff
@@ -0,0 +1,242 @@
+<div class="p-modal" id="contact-modal">
+  <div class="p-modal__dialog" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description">
+    <header class="p-modal__header" style="border-bottom: 0; display: block;">
+      <button class="p-modal__close" aria-label="Close active modal" style="margin-left: -1rem">Close</button>
+      <h2 class="p-modal__title u-sv1" id="modal-title">Tell us about your requirements</h2>
+    </header>
+    <div class="js-pagination js-pagination--1">
+      <p>Tell us more about your needs here and we will send you a detailed report including the required number of nodes for your private cloud and estimated private cloud pricing, including initial CapEx and recurring OpEx costs. We will also help you find the cloud architecture that best suits your requirements.</p>
+      <h3 class="p-heading--four">Instance type</h3>
+      <div class="row u-no-padding">
+        <div class="col-4 u-sv3">
+          <div class="js-formfield">
+            <h3 class="p-heading--five">vCPUs</h3>
+            <input type="number" id="instance-type-vcpus" min="1" max="116" value="2" aria-label="Instance type: vCPUs">
```

Comment:
Oh I thought this value would come from the selection on the previous page.

Reply:
I'm not sure how to grab the variables from the react component to inject them into the contact form, do you know how I could do this?

Reply:
I don't think you can without ejecting from React.

Reply:
I'm still trying to figure out how to do this but all the other comments have been addressed and changes pushed. Do you think we should make another issue with this and the ability to add another instance together? Or I can continue working on it next week

Reply:
This is an architectural issue. Maybe follow up with an issue to address this. I believe you can set the state to window variables with pulls the values when opening the modal using vanilla JS.

Reply:
thanks, managed to figure it out in the end! and did pretty much this ^